datafusion/core/src/physical_optimizer/enforce_distribution.rs

```diff
@@ -1198,37 +1191,25 @@ fn ensure_distribution(
     )
     .map(
         |(mut child, requirement, required_input_ordering, would_benefit, maintains)| {
-            // Don't need to apply when the returned row count is not greater than 1:
```

Comment:
Based on what value is compared with (`batch_size`), I think this should be `batch_size` instead of `1`.